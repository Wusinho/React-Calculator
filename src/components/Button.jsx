/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';

const Squarebtn = (props) => (<button className="square" onClick={props.onClick}>{props.value}</button>);

Squarebtn.propTypes = {
  value: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

Squarebtn.defaultProps = {
  value: '0',
};

export default Squarebtn;
