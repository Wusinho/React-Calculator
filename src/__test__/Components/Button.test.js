import { create } from 'react-test-renderer';

import Button from '../../components/Button';

it('Should renders correctly', () => {
  const elem = create(<Button buttonName="button" handlerClick={() => 'A button'} type="sp" />);
  expect(elem).toMatchSnapshot();
});
