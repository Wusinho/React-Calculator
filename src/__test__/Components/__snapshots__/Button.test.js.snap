// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should renders correctly 1`] = `
<button
  className="square sp"
  classbuttonname="square"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;
